agent/eventhandler/task_handler.go

```diff
@@ -47,13 +47,18 @@ type TaskHandler struct {
 	tasksToEvents map[string]*eventList
 	// tasksToEventsLock for locking the map
 	tasksToEventsLock sync.RWMutex
+	// tasksToContainerStates is used to collect container events
+	// between task transitions
+	tasksToContainerStates     map[string][]api.ContainerStateChange
+	tasksToContainerStatesLock sync.RWMutex
```

Comment:
Can you switch to using a single lock for this struct? Want to avoid the repeat of "deadly embrace" because of multiple locks